src/main/java/edu/utdallas/davisbase/storage/TableFile.java

```diff
@@ -158,23 +157,29 @@ public void appendRow(TableRowBuilder tableRowBuilder) throws IOException {
 
 		file.seek(pageOffset);
 		pageType = file.readByte();
+		if (pageType == 0x0D) {
+			rowId = getnextRowId(file);
+			file.seek(0x01);
+			file.writeInt(rowId);
+		}
 		Boolean splitFlag = false;
```

Comment:
Why is this `Boolean` instead of `boolean`? Nullable booleans should be avoided.

```suggestion
		boolean splitFlag = false;
```